packages/itwinui-css/src/utils/spacing/flex.scss

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+
+@mixin iui-flex {
+  --iui-flex-display: flex;
+  --iui-flex-direction: row;
+  --iui-flex-gap: var(--iui-size-2xs);
```

Comment:
i chose a random default value for gap. we might want to adjust this.

also, we need to find designated gap values for common scenarios:
- vertical gap between inputs/buttons (e.g. forms)
- horizontal gap between inputs/buttons (e.g. toolbars, forms, button bars)
- horizontal gap between text and icon
- vertical gap between label and sublabel

Reply:
@FlyersPh9

Reply:
- **vertical gap between inputs/buttons (e.g. forms)**
We are using `var(--iui-size-s)` for vertical gap between inputs on [input](https://itwin.github.io/iTwinUI/css/input.html), [information panel](https://itwin.github.io/iTwinUI/css/information-panel.html), & [column filter](https://itwin.github.io/iTwinUI/css/column-filter.html).

- **horizontal gap between inputs/buttons (e.g. toolbars, forms, button bars)**
We are using `var(--iui-size-xs)` for horizontal gap between buttons in [dialog](https://itwin.github.io/iTwinUI/css/dialog.html) & [information panel buttons above the table](https://itwin.github.io/iTwinUI/css/information-panel.html).

- **horizontal gap between text and icon**
We are using `var(--iui-size-2xs)` for horizontal gap between icon & text in [input status message](https://itwin.github.io/iTwinUI/css/input), but `var(--iui-size-xs)` for the horizontal gap between icon & text in [table's name column](https://itwin.github.io/iTwinUI/css/table.html).  If we want to standardize I'd lean more towards `var(--iui-size-xs)`, but maybe it's ok for them to be different due to the different font sizes.

- **vertical gap between label and sublabel**
It would appear we have no gap used in [header](https://itwin.github.io/iTwinUI/css/header.html), [menu](https://itwin.github.io/iTwinUI/css/menu.html), & [expandable block](https://itwin.github.io/iTwinUI/css/expandable-block.html).  Altho in the expandable block it looks like we are hard coding a height for the text & not relying on font size to set the height.

I'd say the first 3 seem safe to standardize, but the vertical gap between label & sublabel we should investigate what we should use.

Reply:
I've changed the default gap to `--iui-size-xs`. This will automatically handle cases 2 and 3.

For case 1, I've added `--iui-spacing-form-vertical` to the global tokens. The name and location of this variable is debatable, but as far as the flex component is concerned, it can be used like this:
```jsx
<Flex gap='form-vertical'>
  ...
</Flex>
```

This will still not handle content shift when a status message appears, which makes me think we should exclude this variable from this PR and try to tackle these two things together.